importer/tasks.py

```diff
@@ -133,22 +154,9 @@ def get_collection_items(collection_url):
 
     items = []
     current_page_url = collection_url
-    max_attempts = 10
 
     while current_page_url:
-        attempts = 0
-
-        while attempts < max_attempts:
-            resp = requests.get(current_page_url)
-
-            if resp.status_code != 429 and resp.status_code != 503:
-                break
-
-            # If rate limited or service unavailable, wait and try again
-            time.sleep(2 * attempts)
-            attempts = attempts + 1
-
-        resp.raise_for_status()
+        resp = requests_retry_session().get(current_page_url)
```

Comment:
We could hoist this session up to the top so we reuse it for all of the URLs but I'm not sure how much we care about that given that we're actually trying to slow things down a bit.

Reply:
I thought of that, but then I realized the other requests are relying on the celery backoff / retry mechanism. I actually was able to create that scenario in my testing and I think it's better to let celery handle it for the asynchronous idempotent stuff, rather than allow the delayed request to occupy a celery worker thread.

I put a lot of thought into whether / how we should use the Django cache for the responses of the collection pages as well - I don't think that gets us much unless we build a retry function for the main import job. You can currently "manually" retry by going through the Import Items form again and putting in the same collection URL, in which case the cached results would give some benefit, but I believe the main problem we're experiencing is that we're rate limited as we page through collection results - and this solution solves that by slowing down enough to get through all the pages. I think I'll put the cache stuff in anyway, for the manual case and just in case we build that main retry function.

Reply:
Oh, yes, I was definitely thinking that as part of this we update https://github.com/LibraryOfCongress/concordia/blob/8f7525de4ce0ca901d116b4f04a23a4c8f6ecb66/importer/tasks.py#L267-L268 to enable automatic retries, perhaps with just a simple multi-minute time delay.